### PR TITLE
[WIP]add more $vars

### PR DESCRIPTION
**Pull request by enoch85**
leave theese vars in:
export PHPMYADMIN_CONF
export NCUSER
CLEARBOOT=$(dpkg -l linux-* | awk '/^ii/{ print $2}' | grep -v -e "$(uname -r | cut -f1,2 -d"-")" | grep -e "[0-9]" | xargs sudo apt -y purge)

### CODE REVIEW

**Comment by morph027** [comment]
Let me know when you squashed all vars...i (hopefully) should have a green ci run then ;)

**Comment by morph027** [comment]
Jips, i do...finally all shellcheck errors gone ;)

https://travis-ci.org/morph027/vm/builds/218418063

File: static/security.sh [hunk 1]
```diff
@@ -10,11 +10,11 @@ true
...
-SPAMHAUS=/etc/spamhaus.wl
-ENVASIVE=/etc/apache2/mods-available/mod-evasive.load
-APACHE2=/etc/apache2/apache2.conf
-
-set -e
+# Check for errors + debug code and abort if something isn't right
+# 1 = ON
+# 0 = OFF
+DEBUG=0
+debug_mode
```
Comment (enoch85) [inline static/security.sh]: here

**Comment by enoch85** [comment]
@morph027 Okay, that should be it.

Added `debug_mode` instead if the complete code as a function. Could you please update all scripts with that?

**Comment by enoch85** [comment]
Another Idea I got is to add a function to ALWAYS check all $vars if they are empty. That would be really helpful when debugging.

**Comment by morph027** [comment]
Jip!

**Comment by morph027** [comment]
> Another Idea I got is to add a function to ALWAYS check all $vars if they are empty. That would be really helpful when debugging.

Check if they are set?

**Comment by enoch85** [comment]
@morph027 if they are empty (or set, idk the specific wording)

File: nextcloud-startup-script.sh [hunk 2]
```diff
@@ -11,16 +11,9 @@ true
 # 0 = OFF
 DEBUG=0
 
-WWW_ROOT=/var/www
-NCDATA=/var/ncdata
-CLEARBOOT=$(dpkg -l linux-* | awk '/^ii/{ print $2}' | grep -v -e "$(uname -r | cut -f1,2 -d"-")" | grep -e "[0-9]" | xargs sudo apt -y purge)
-PHPMYADMIN_CONF="/etc/apache2/conf-available/phpmyadmin.conf"
 export PHPMYADMIN_CONF
```
Comment (morph027) [inline nextcloud-startup-script.sh]: We don't need ```export PHPMYADMIN_CONF``` anymore when included in lib.sh

File: nextcloud-startup-script.sh [hunk 3]
```diff
@@ -11,16 +11,9 @@ true
...
-STATIC="https://raw.githubusercontent.com/nextcloud/vm/master/static"
-LETS_ENC="https://raw.githubusercontent.com/nextcloud/vm/master/lets-encrypt"
-NCPASS=nextcloud
-NCUSER=ncadmin
 export NCUSER
```
Comment (morph027) [inline nextcloud-startup-script.sh]: Same here... ```export $NCUSER``` should be available though lib.sh

**Comment by enoch85** [comment]
So shellchek checks if vars are empty? Cool stuff!

**Comment by enoch85** [comment]
OK, if you are good with this then please merge and work on the stuff I added in the list in that old PR.

Thanks!

**Comment by enoch85** [comment]
@morph027 Doesn't https://cdn.rawgit.com/morph027/vm/master/lib.sh update automatically? Just tried the script and all the $vars I added are missing...

**Comment by morph027** [comment]
Argh, probably a caching issue

http://rawgit.com/

> Use a specific tag or commit hash in the URL (not a branch). Files are cached permanently based on the URL. Query strings are ignored.

But we could also use the real github file in this case...I'll fix this.

**Comment by enoch85** [comment]
**ERRORS**
- [ ] The collabora whiptail comes up in every script
- [ ] mysql_password.txt isn't found (it's not created when you install) The error is whenit tries to cat in the installation script

**Comment by enoch85** [comment]
@morph027 Could you have a look please?

**Comment by morph027** [comment]
Already on it...

**Comment by enoch85** [comment]
Thanks!

**Comment by morph027** [comment]
Collabora stuff should be working now...

**Comment by enoch85** [comment]
Marked as solved

**Comment by enoch85** [comment]
Updated the comment

**Comment by enoch85** [comment]
Please also add the debug_mode everywhere

**Comment by enoch85** [comment]
In the top of all scripts so that we get the curl as well... @morph027

**Comment by enoch85** [comment]
**Things to do**
- [x] https://github.com/morph027/vm/pull/13#issuecomment-291012622 Fix find command 
- [ ] https://github.com/morph027/vm/pull/13#discussion_r109451589 fix a while loop 
- [x] Don't update major versions (I will do it)
- [ ] https://github.com/morph027/vm/pull/13#discussion_r109454550 while loop maybe?
- [ ] https://github.com/morph027/vm/pull/13#discussion_r109459050 make sure change_mysql_pass finishes with 0, then continue
- [ ] Improve the lib.sh
- [x] Fix the collabora script https://github.com/morph027/vm/pull/14#issuecomment-291054487
- [ ] Make webmin | contacts | calendar separate scripts and put all app scripts in a seperate /apps folder
- [ ] call the phpmyadmin script as a function instead (https://github.com/morph027/vm/blob/global-functions-and-vars/nextcloud-startup-script.sh#L536)
- [ ] Make all [these](https://github.com/morph027/vm/blob/20-increase-code-quality/nextcloud-startup-script.sh#L130) a function instead so it can be easily reproduced when adding a new script 
- [ ] Fix the ncadmin script https://github.com/morph027/vm/pull/13#issuecomment-291231759 (chown issue)
- [ ] Add a spinner all the stuff that takes a while: https://github.com/morph027/vm/pull/12 and improve the function even more
- [ ] Add ALL vars in lib.sh and comment which script they belong to if they are specific

**Comment by enoch85** [comment]
@morph027 Please check the remaining issues here. Thanks!

**Pull request merged**